common/djangoapps/student/models.py

```diff
@@ -933,29 +933,13 @@ def clear_lockout_counter(cls, user):
         except ObjectDoesNotExist:
             return
 
-    def __repr__(self):
-        """Repr -> LoginFailures(username, count, date)"""
-        date_str = '-'
-        if self.lockout_until is not None:
-            date_str = self.lockout_until.isoformat()
-
-        return u'LoginFailures({username}, {count}, {date})'.format(
-            username=six.text_type(self.user.username, 'utf-8'),
-            count=self.failure_count,
-            date=date_str
-        )
-
     def __str__(self):
         """Str -> Username: count - date."""
-        date_str = '-'
-        if self.lockout_until is not None:
-            date_str = self.lockout_until.isoformat()
-
-        return u'{username}: {count} - {date}'.format(
-            username=six.text_type(self.user.username, 'utf-8'),
+        return six.text_type('{username}: {count} - {date}'.format(
+            username=self.user.username,
```

Comment:
I don't think this fixes the problem.  This will still try to format a unicode string into a bytestring on Python 2, which will fail with a UnicodeEncodeError if the username has non-ascii characters in it.

Reply:
Right, I just removed this casting and left the top one for unicode.